### PR TITLE
Add min max count assertions

### PR DESCRIPTION
**Pull request by Taluu**

### CODE REVIEW

File: lib/Assert/Assertion.php [hunk 1]
```diff
@@ -1782,6 +1788,56 @@ public static function count($countable, $count, $message = null, $propertyPath
...
+     * @param string|null      $message
+     * @param string|null      $propertyPath
+     *
+     * @return bool
+     */
+    public static function minCount($countable, $count, $message = null, $propertyPath = null)
+    {
+        if ($count > \count($countable)) {
+            $message = \sprintf(
+                static::generateMessage($message ?: 'List should have at least %d elements, but only has %d elements.'),
```
Comment (rquadling) [inline lib/Assert/Assertion.php]: Grammar matching.  "List should have no [more|fewer] than %d elements, but has %d elements".

Comment (Taluu) [reply lib/Assert/Assertion.php]: I wasn't sure about the grammar, so I'll make the change 👍

File: lib/Assert/Assertion.php [hunk 2]
```diff
@@ -1782,6 +1788,56 @@ public static function count($countable, $count, $message = null, $propertyPath
...
+     * @param string|null      $message
+     * @param string|null      $propertyPath
+     *
+     * @return bool
+     */
+    public static function minCount($countable, $count, $message = null, $propertyPath = null)
+    {
+        if ($count > \count($countable)) {
+            $message = \sprintf(
+                static::generateMessage($message ?: 'List should have no fewer than least %d elements, but has %d elements.'),
```
Comment (rquadling) [inline lib/Assert/Assertion.php]: List should have no fewer than %d elements, but has %d elements

File: lib/Assert/Assertion.php [hunk 3]
```diff
@@ -1782,6 +1788,56 @@ public static function count($countable, $count, $message = null, $propertyPath
         return true;
     }
 
+    /**
+     * Assert that the countable has at least $count elements.
```
Comment (rquadling) [inline lib/Assert/Assertion.php]: Assert that the countable has no fewer than $count elements.

File: lib/Assert/Assertion.php [hunk 4]
```diff
@@ -1782,6 +1788,56 @@ public static function count($countable, $count, $message = null, $propertyPath
...
+     * @param string|null      $message
+     * @param string|null      $propertyPath
+     *
+     * @return bool
+     */
+    public static function maxCount($countable, $count, $message = null, $propertyPath = null)
+    {
+        if ($count < \count($countable)) {
+            $message = \sprintf(
+                static::generateMessage($message ?: 'List should have no more than %d elements, but has %d elements.'),
```
Comment (rquadling) [inline lib/Assert/Assertion.php]: This one is right.

**Comment by rquadling** [comment]
Either `at least` and `at most`, or `no fewer` and `no more`.

**Comment by rquadling** [comment]
Sorry. Clicked close rather than comment.

**Comment by Taluu** [comment]
Np. :}

I went for `at least` and `at most` then.

**Comment by rquadling** [comment]
And some unit tests please. Prove that it all works.

**Comment by Taluu** [comment]
I also thought about adding tests, but I saw that none of the current assertions have those, so I wasn't sure on how to test it.

Should I create a new assertion test case then ?

**Comment by rquadling** [comment]
> I also thought about adding tests, but I saw that none of the current assertions have those, so I wasn't sure on how to test it.
> 
> Should I create a new assertion test case then ?

https://github.com/beberlei/assert/blob/master/tests/Assert/Tests/AssertTest.php

**Comment by rquadling** [comment]
We currently have 100% code coverage, so, yes, tests please!!!

**Comment by rquadling** [comment]
I do need to break the tests up into logical/associated assertions to make this all a little easier to see.

**Comment by Taluu** [comment]
`AssertTest` should also be renamed (before splitting it ?) to `AssertionTest` so that it is quicker and easier to map it.

I'll add the tests then and do the same for the other PRs.

**Comment by Taluu** [comment]
Tested added everywhere. :}

**Comment by rquadling** [comment]
Taken your code and incorporated the additional countable types into the doc block typehint and unit tests.
